core/fields/Field_Rich.php

```diff
@@ -236,17 +304,77 @@ public function display() {
-							let new_title = window.prompt('Enter TITLE text: ',active_image.title);
-							if (new_alt) {
-								active_image.alt = new_alt;
-							}
-							if (new_alt) {
-								active_image.title = new_title;
-							}
-							active_image.classList.remove('rich_image_active');
-							// push updated content to textarea
-							active_image.closest('.control').querySelector('textarea').value = active_image.closest('.editor').innerHTML;
+
+							// get current image props
+							var title = active_image.title;
+							var alt = active_image.alt;
+
+							// display modal
+							var modal = document.createElement('div');
+							modal.innerHTML = `
+								<div class="modal-background"></div>
+								<div class="modal-content">
+									<div class="box">
+										
+										<div class="field">
+											<label class="label">Alt Text</label>
+											<div class="control">
+												<input id="new_alt_text" class="input" placeholder='' type="text" value="` + alt + `">
+												<p class="help">Alternative text for the visually impaired, will also display when the browser cannot render the image.</p>
+											</div>
+										</div>
+									
+										<div class="field">
+											<label class="label">Image Title</label>
+											<div class="control">
+												<input id="new_image_title" class="input" type="text" value="` + title + `">
+											</div>
+											<p class="help">Title will appear in a tooltip on hover of the image. It can also be used to provide a broader description than the alt text.</p>
+
+										</div>
+
+										<button id="modal_save" class="button is-primary">Add</button>
+										<button id="modal_cancel" class="button is-warning">Cancel</button>
+
+									</div>
+								</div>
+							`;
+							modal.classList = "modal";
+							document.body.appendChild(modal);
+							modal.classList.add("is-active");
+
+							// handle cancel
+							document.getElementById('modal_cancel').addEventListener('click',function(e){
+
+								// close modal
+								var modal = e.target.closest('.modal');
+								modal.parentNode.removeChild(modal);
+
+							});
+
+							// handle save
+							document.getElementById('modal_save').addEventListener('click',function(e){
+								
+								// set to image
+								let new_alt = document.getElementById('new_alt_text').value;
+								let new_title = document.getElementById('new_image_title').value;
+								if (new_alt) {
+									active_image.alt = new_alt;
+								}
+								if (new_title) {
+									active_image.title = new_title;
+								}
+								active_image.classList.remove('rich_image_active');
+								
+								// push updated content to textarea
+								active_image.closest('.control').querySelector('textarea').value = active_image.closest('.editor').innerHTML;
+
+								// close modal
+								var modal = event.target.closest('.modal');
+								modal.parentNode.removeChild(modal);
+
+							});
```

Comment:
just saw this, see previous comment, really be better to have one source of truth rather than trying to make changes to 2 copies